blog/.vuepress/plugins/sidebar/create.js

```diff
@@ -40,18 +30,11 @@ module.exports = filesExtractedDatas => {
           }) => {
             
             const sidebar = ${JSON.stringify(sidebar)};
-            const navBarAddition = ${JSON.stringify(navBarAddition)}
-            const events = ${JSON.stringify(Events)}
-            const tags = ${JSON.stringify(Tags)}
+            const archives = ${JSON.stringify(Archives)};
             if (!siteData.themeConfig['sidebar']) siteData.themeConfig['sidebar'] = {}
-            siteData.themeConfig.sidebar['/archives/'] = sidebar
+            siteData.themeConfig.sidebar['/archives/'] = sidebar;
             if (!siteData.themeConfig.nav) siteData.themeConfig.nav = []
-            siteData.themeConfig.nav = [
-              events,
-              tags,
-              navBarAddition,
-              ...siteData.themeConfig.nav
-            ];
+            siteData.themeConfig.nav[0].items[2] = archives
```

Comment:
This could be improved with a recursive search in the `nav` object so we can later on move the `Archives` anywhere in the hierarchy:

Something like:
```
siteData.themeConfig.nav = siteData.themeConfig.nav.reduce(function inflateNav(acc, curr) {
  // TODO: move this condition in an map if we need to inflate more than Archives in the future
  if (curr.text === 'Archives') {
    curr.link = archives
  }

  if (curr.items) {
    curr.items = curr.items.reduce(inflateNav, []);
  }

  acc.push(curr);

  return acc;
}, [])
```

Reply:
definitely agree with what you say. It could be an improvement. But we will change the creation logic. Look at my features issues i created. We should first do them and adding test for those parts. But anyway it's a good improvment 👍